src/aiori-IME.c

```diff
@@ -371,28 +399,32 @@ static int IME_RmDir(const char *path, IOR_param_t *param)
 /*
  * Perform stat() through the IME interface.
  */
-static int IME_Stat(const char *path, struct stat *buf, IOR_param_t *param)
+int IME_Stat(const char *path, struct stat *buf,
+             aiori_mod_opt_t *module_options)
 {
-    (void)param;
+    (void)module_options;
 
     return ime_native_stat(path, buf);
 }
 
 /*
  * Use IME stat() to return aggregate file size.
  */
-static IOR_offset_t IME_GetFileSize(IOR_param_t *test, MPI_Comm testComm,
-                                    char *testFileName)
+IOR_offset_t IME_GetFileSize(aiori_mod_opt_t *test, MPI_Comm testComm,
+                             char *testFileName)
 {
+        if (hints->dryRun)
+                return 0;
+
         struct stat stat_buf;
         IOR_offset_t aggFileSizeFromStat, tmpMin, tmpMax, tmpSum;
 
-        if (ime_native_stat(testFileName, &stat_buf) != 0) {
-                ERR("cannot get status of written file");
-        }
+        if (ime_native_stat(testFileName, &stat_buf) != 0)
+                ERRF("cannot get status of written file %s", testFileName);
+
         aggFileSizeFromStat = stat_buf.st_size;
 
-        if (test->filePerProc) {
+        if (hints->filePerProc) {
```

Comment:
This code part can be removed.

Reply:
thanks, I'm going to fix that.

Reply:
Seems this is not done yet. I'm getting warnings if I remove the reduction.

```
WARNING: Expected aggregate file size       = 17179869184.
WARNING: Stat() of aggregate file size      = 4294967296.
WARNING: Using actual aggregate bytes moved = 17179869184.
```

Reply:
Could you specify which options you have used?
The code responsible is in ior.c:327 and following. That was used in POSIX before and I wonder how that could have happened here.

Reply:
It does not work with FPP as in ior.c:327 the reduction only concerns aggFileSizeFromXfer. With FPP it can only be compared to aggFileSizeFromStat if there was a previous reduction on all files before. For now this is done in each aiori backend.

Reply:
Now I see the confusion: could you check with the branch: feature-cleanup-aiori
That one contains some further changes regarding the file size.
I merge this pull now, the delta to the other branch should be minimal.